app/code/Magento/ConfigurableProduct/Model/Plugin/ProductRepositorySave.php

```diff
@@ -60,7 +60,7 @@ public function beforeSave(
         ProductInterface $product,
         $saveOptions = false
     ): array {
-        $result[] = $product;
+        $result = [$product, $saveOptions];
```

Comment:
It looks like we're not expecting to change any arguments for the "save" method, so I think it's better to replace the return type with "void" and don't change the method arguments.